src/workflow-handler.ts

```diff
@@ -127,64 +130,47 @@ export class WorkflowHandler {
       return this.workflowRunId;
     }
     try {
-      core.debug('Get workflow run id');
+      const workflowId = await this.getWorkflowId();
```

Comment:
Too much code here, this is not clear.
Please split into several methods for quicker review with explicit names.
For example :
```
private async findAllWorkflowRuns() {
  try {
    const workflowId = await this.getWorkflowId();
    const response = await this.octokit.rest.actions.listWorkflowRuns({
        owner: this.owner,
        repo: this.repo,
        workflow_id: workflowId,
        event: 'workflow_dispatch',
        created: `>=${new Date(this.triggerDate).toISOString()}`
      });

      debug('List Workflow Runs', response);

      return response.data.workflow_runs
  } catch (e) {
     debug(...)
     throw new Error(`Failed to list workflow runs. Cause: ${e}`) 
  }
}